### ISSUE TITLE
Investigate R28980 & R20574 sample swap

### ISSUE THREAD

**Issue by mbhall88** [mbhall88/head_to_head_pipeline#68]
This pair has a COMPASS distance of 0 and a `bcftools` distance of 27. It is the cause of the only FN for SNP thresholds 0, 2 & 5 and one of the two causes for SNP threshold 12.

**Comment by mbhall88** [comment]
R20574 seems to be the cause of this discrepancy. The reason for this is that R20260 has total agreement between COMPASS and bcftools at these 27 sites, but R20574 has differences between COMPASS and bcftools at **all** of these sites.

The sample has high depth at all of these positions, shows no signs of  HETs, shows no signs of strand bias,  The lineage call for the nanopore data is also the same as the Illumina data.

I'm at a bit of a loss as to what else to do other than just :shrug:

**Comment by mbhall88** [comment]
For future reference, the positions where the 27 differences occur is 

```
[7581, 131114, 143463, 637419, 737014, 761109, 761154, 828219, 886669, 1055774, 1424975, 1471473, 1473245, 1673422, 2205857, 2215433, 2287683, 2289201, 2322806, 2333097, 2605149, 2976245, 3204704, 3511367, 3665012, 4247430, 4248001]
```

The bcftools VCF is on noah at `/hps/nobackup/research/zi/projects/tech_wars/analysis/baseline_variants/nanopore/filtered_snps/south_africa/R20574.snps.filtered.bcf`  
The COMPASS VCF is at `/hps/nobackup/research/zi/projects/tech_wars/analysis/baseline_variants/illumina/gvcfs/R20574.compass.vcf.gz`

**Comment by iqbal-lab** [comment]
How about reopen this and transfer ownership to me and Martin for a couple of days, we can look on mon/tues

**Comment by iqbal-lab** [comment]
Any chance you couldput paths to bams here, so we can look at the pileups?

**Comment by mbhall88** [comment]
That would be very much appreciated.  
I don't have BAMs (I delete them after I have created the pileup). But I have a pileup from [`bcftools mpileup`](https://samtools.github.io/bcftools/bcftools.html#mpileup) using option `--ignore-overlaps` which Is on noah at `/hps/nobackup/research/zi/projects/tech_wars/analysis/baseline_variants/nanopore/pileups/south_africa/R20574.pileup.bcf` this pileup is then used to call SNPs by `bcftools call`

I did just create a BAM for this too though if you want it `/hps/nobackup/research/zi/projects/tech_wars/analysis/baseline_variants/tmp/R20574.bam`

**Comment by iqbal-lab** [comment]
Does Martin have access to this repo btw? Will take a look at this with him, either Monday afternoon or Tuesday first thing

**Comment by mbhall88** [comment]
It's a public repo, so he should be able to see all of this.

**Comment by mbhall88** [comment]
I think I have figured out what is happening here!

**tl;dr** there was a sample swap between R20574 and R28980

I actually realised there was something weird going on here when looking at the differences between the COMPASS and bcftools clusters. I was looking at the samples that had poor recall and precision cluster-wise between the two. It is best illustrated with a drawing

![IMG_1827](https://user-images.githubusercontent.com/20403931/111307020-a40ce880-86a4-11eb-81f8-e2c61ab4c300.JPEG)

In red boxes are the samples. The Illumina distance between 2 samples is represented with a blue line and number, and nanopore with black. I've put a red square around the noticeably discrepant distances.  
So we see there is something funny going on between the trio of R20260, R28980, and R20574 - two of which are the samples in the title of this issue.  

When I look at the positions with SNP differences between R20260 and R28980 on Illumina and compare those positions to the differences between R20260 and R20574 on nanopore they overlap in the following way:
- The shared positions (26) have the same ALTs and GT calls
- The positions in Illumina not in nanopore (7) are all filtered in nanopore, with the GT calls matching the Illumina
- The positions in Nanopore not in Illumina (1) are filtered in Illumina, with the GT calls matching the nanopore

The next thing I did was look at the pairwise differences between the Illumina and Nanopore consensus sequences for these three samples. The `_b` after the sample name indicates bcftools (nanopore) and the `_c` indicates COMPASS (Illumina)

```
R20260_c x R20574_c: 0
R20260_c x R28980_c: 34
R20260_c x R20260_b: 0
R20260_c x R20574_b: 28
R20260_c x R28980_b: 0
R20574_c x R28980_c: 34
R20574_c x R20260_b: 0
R20574_c x R20574_b: 28
R20574_c x R28980_b: 0
R28980_c x R20260_b: 33
R28980_c x R20574_b: 0
R28980_c x R28980_b: 32
R20260_b x R20574_b: 27
R20260_b x R28980_b: 0
R20574_b x R28980_b: 26
```

Admittedly this melted my brain for a little while. But there are four distances that explain the whole thing

```
R20574_c x R20574_b: 28
R28980_c x R28980_b: 32
R28980_c x R20574_b: 0
R20574_c x R28980_b: 0
```

### Conclusion

I'm getting to the limits of how much my brain can handle in a single sitting - so correct me if this is wrong - but is it actually possible to know whether it was the Illumina or nanopore that got swapped?

I'll forward this on to Anastasia and see if she can shed any light.

**Comment by mbhall88** [comment]
Close pairwise distance dotplot **before** sample swap detection

![bokeh_plot](https://user-images.githubusercontent.com/20403931/111727095-fa9d4100-88b5-11eb-859f-07bac9bc3d9b.png)

And **after** swapping the Nanopore data for the two samples

![bokeh_plot](https://user-images.githubusercontent.com/20403931/111727160-156fb580-88b6-11eb-8a6f-b1a9bab177c9.png)

We get a sizeable improvement in correlation coefficient!  

Results from clustering will be added to #65 

*Note: this is not quite final as in order to speed up this analysis I ran `bcftools mpileip` with no per-base alignment quality (BAQ). We know from #38 that this results in slight worse precision. Once the longer, more accurate `bcftools mpileup` with BAQ is finished I will update the plots*

**Comment by mbhall88** [comment]
The DST results confirm that this was indeed a Nanopore sample swap. If we hadn't swapped the labelling on the files, the mykrobe Illumina and Nanopore predictions would have been quite different, and the nanopore ones would have disagreed with the available DST for these samples
